### PR TITLE
Translate new construction of NamedTuple

### PR DESCRIPTION
**Pull request by KDr2**
Fix https://github.com/TuringLang/Turing.jl/issues/1873

### CODE REVIEW

File: src/tapedfunction.jl [hunk 1]
```diff
@@ -437,7 +446,7 @@ function translate!!(var::IRVar, line::Expr,
     end
 end
 
-function translate!!(var, line, bindings, ir)
+function translate!!(var, line, bindings, i, ir)
```
Comment (yebai) [inline src/tapedfunction.jl:449]: What's this additional argument `i'?

**Review by yebai — APPROVED** [review]
Thanks, @KDr2 - it looks good other than the minor clarification question below.

File: src/tapedfunction.jl (hunk 1 above)
Comment (KDr2) [inline src/tapedfunction.jl:449]:
```suggestion
function translate!!(var, line, bindings, isconst, ir)
```

Comment (KDr2) [reply src/tapedfunction.jl:449]: the signature of `translate!!` changed in a past PR, I forgot to update this method then.

**Pull request merged**